pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -390,28 +390,41 @@ def on_http_proxy_checkbox_toggled(self, checkbox):
             # the subscription request structure, but keep it in the entries in case
             # the user tries to show them again before next spoke entry clears them
             self._subscription_request.server_proxy_hostname = ""
-            self._subscription_request.server_proxy_posr = -1
+            self._subscription_request.server_proxy_port = -1
             self._subscription_request.server_proxy_user = ""
-            self._subscription_request.server_proxy_password = ""
+            self._subscription_request.server_proxy_password.set_secret(None)
 
     def on_http_proxy_location_entry_changed(self, editable):
-        hostname = ""
-        port = -1  # not set == -1
-        entered_text = editable.get_text()
-        if entered_text:
-            proxy_obj = ProxyString(url=entered_text)
+        # Incorrect hostnames, including empty strings, will
+        # throw an exception we need to catch and switch
+        # to defaults. This can happen often as the user
+        # types the hostname to the field.
+        try:
+            proxy_obj = ProxyString(url=editable.get_text())
             hostname = proxy_obj.host
             if proxy_obj.port:
                 # the DBus API expects an integer
                 port = int(proxy_obj.port)
+        except ProxyStringError:
+            hostname = ""
+            port = -1  # not set == -1
+        # set the resulting values to the DBus structure
         self.subscription_request.server_proxy_hostname = hostname
         self.subscription_request.server_proxy_port = port
 
     def on_http_proxy_username_entry_changed(self, editable):
         self.subscription_request.server_proxy_user = editable.get_text()
 
     def on_http_proxy_password_entry_changed(self, editable):
-        self.subscription_request.server_proxy_password = editable.get_text()
+        password = editable.get_text()
+        # check if password field is non empty
+        if password:
+            # there is a password, set it
+            self.subscription_request.server_proxy_password.set_secret(password)
+        else:
+            # there is no password, clear the password (if any) by setting
+            # the secret to None
+            self.subscription_request.server_proxy_password.set_secret(None)
```

Comment:
You can replace the `if-else` block with:

`self.subscription_request.server_proxy_password.set_secret(password or None)`

Reply:
Good idea, changed. :)